src/backend/pipeline/cont_scheduler.c

```diff
@@ -46,7 +46,9 @@
 
 #define MAX_PROC_TABLE_SZ 16 /* an entry exists per database */
 #define INIT_PROC_TABLE_SZ 4
-#define TOTAL_SLOTS (continuous_query_num_workers + continuous_query_num_combiners)
+
+/* extra +1 slot is for the adhoc query */
+#define TOTAL_SLOTS (continuous_query_num_workers + continuous_query_num_combiners + 1)
```

Comment:
It seems that the only use an adhoc proc has for the existing process metadata infrastructure is to read tuples the same way that continuous views currently do. We'll probably need to rework that slightly so that adhocs procs can't interfere with regular combiner/worker procs, so we should be able to just do away with any changes to this file.